cmd/ingester/app/consumer/consumer.go

```diff
@@ -109,9 +110,15 @@ func (c *Consumer) Close() error {
 
 func (c *Consumer) handleMessages(pc sc.PartitionConsumer) {
 	c.logger.Info("Starting message handler", zap.Int32("partition", pc.Partition()))
-	c.partitionsHeld.Inc(1)
-	defer c.partitionsHeld.Inc(-1)
 	c.partitionMapLock.Lock()
+	c.partitionsHeld++
+	c.partitionsHeldGauge.Update(c.partitionsHeld)
+	defer func() {
+		c.partitionMapLock.Lock()
+		c.partitionsHeld--
+		c.partitionsHeldGauge.Update(c.partitionsHeld)
+		c.partitionMapLock.Unlock()
+	}()
```

Comment:
I recommend moving this after L123 (after unlock) and combining L124-125 into the same `defer func()`

Reply:
Makes sense, amended.